### ISSUE TITLE
Обработка ошибок

### ISSUE THREAD

**Issue by fonrit** [Tinkoff/AcquiringSdkAndroid#149]
Можно ли классифицировать полученный exeption для персональной обработки ?
В документации сказано что часть ошибок обрабатываются в самом sdk, остальные придется обрабатывать самим.
Не понятно как из AcquiringSdkException достать статус ошибки AUTH_FAIL для персональной обработки

exeption ru.tinkoff.acquiring.sdk.exceptions.AcquiringSdkException: PaymentState = AUTH_FAIL

**Comment by IlnarH** [comment]
Может, просто достать статус из `message` ошибки?

**Comment by fonrit** [comment]
Это да, а вот **AcquiringSdkException** возвращает только статус (Из этой страницы https://www.tinkoff.ru/kassa/develop/api/payments) который нужно обрабатывать самим, не может ли он вернуть клиенто ориентированную ошибку как **AcquiringApiException**:

**message**: Попробуйте повторить попытку позже Не получилось оплатить. Воспользуйтесь другой картой или обратитесь к продавцу

**Comment by IlnarH** [comment]
Т.е. имеется в виду получение человекочитаемого сообщения от бэка для клиента?

**Comment by fonrit** [comment]
Да. AcquiringApiException возвращает текст и код ошибки, с ним нет проблем кроме того что некоторый текст довольно кривой. А вот из AcquiringSdkException можно вытащить только message, который возвращается в таком виде "PaymentState = AUTH_FAIL", я могу сделать обработчик этих состояний локально, но хочется понимать  подобная текстовка "PaymentState = AUTH_FAIL" будет возвращаться всегда с AcquiringSdkException, или он может вдруг вернуть уже текстовку с описанием ошибки

**Comment by IlnarH** [comment]
Не могли бы вы приложить логи, включающие запрос и ответ при получении этой ошибки?

**Comment by fonrit** [comment]
Речь идет о исключении которое выбрасывает AcquiringSdkException или ошибки из AcquiringApiException ?

**Comment by IlnarH** [comment]
Речь о запросе (насколько я понимаю, `GetState`), который возвращает ответ со статусом платежа `AUTH_FAIL`, т.е. network логи в которых присутствуют строки вида `=== Sending request` и `=== Got server response:`.

**Comment by fonrit** [comment]
Я почему то вижу логи только с теста, с прода есть скрин дебага 

**Тест:**

https://rest-api-test.tinkoff.ru/v2/GetState

Response: 
{
  "Success": true,
  "ErrorCode": "1014",
  "Message": "Карта недействительна",
  "TerminalKey": "TestSDK",
  "Status": "REJECTED",
  "PaymentId": "4072210335",
  "OrderId": "1323658787",
  "Params": [
    {
      "Key": "Route",
      "Value": "ACQ"
    },
    {
      "Key": "Source",
      "Value": "cards"
    }
  ],
  "Details": "Неправильные реквизиты — проверьте их или воспользуйтесь другой картой",
  "Amount": 1620
}

**Прод:**
<img width="768" alt="Снимок экрана 2022-12-13 в 18 47 06" src="https://user-images.githubusercontent.com/28973635/207380801-da085c8b-0276-4b12-b402-36c9d092efac.png">

Судя по респонсу на тесте getState может вернуть более детальную информацию о ошибке, но толка от этого нет так как она не передается дальше в **ThreeDsViewModel**:
<img width="991" alt="Снимок экрана 2022-12-13 в 18 56 10" src="https://user-images.githubusercontent.com/28973635/207381563-a962e7a9-a226-48f4-a4b4-e687f887e2c0.png">
Вы просто возвращаете message со статусом "PaymentState = AUTH_FAIL"

**Comment by fonrit** [comment]
Было хорошо если AcquiringSdkException возвращал message (Опционально) и state, чтобы можно было локально Обработать ошибку в случае необходимости

**Comment by fonrit** [comment]
Можете пожалуйста ответить

**Comment by IlnarH** [comment]
Думаю, можно реализовать что-то подобное, но не в ближайшее время; скорее всего вместе с редизайном (ориентировочно - Q1 2023).
